subcommand/connect-init/command_test.go

```diff
@@ -0,0 +1,181 @@
+package connectinit
+
+import (
+	"github.com/hashicorp/consul-k8s/consul"
+	"github.com/hashicorp/consul-k8s/subcommand/common"
+	"github.com/hashicorp/consul/api"
+	"github.com/mitchellh/cli"
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+)
+
+const testAuthMethod = "consul-k8s-auth-method"
+const loginResponse = `{
+  "AccessorID": "926e2bd2-b344-d91b-0c83-ae89f372cd9b",
+  "SecretID": "b78d37c7-0ca7-5f4d-99ee-6d9975ce4586",
+  "Description": "token created via login",
+  "Roles": [
+    {
+      "ID": "3356c67c-5535-403a-ad79-c1d5f9df8fc7",
+      "Name": "demo"
+    }
+  ],
+  "ServiceIdentities": [
+    {
+      "ServiceName": "example"
+    }
+  ],
+  "Local": true,
+  "AuthMethod": "minikube",
+  "CreateTime": "2019-04-29T10:08:08.404370762-05:00",
+  "Hash": "nLimyD+7l6miiHEBmN/tvCelAmE/SbIXxcnTzG3pbGY=",
+  "CreateIndex": 36,
+  "ModifyIndex": 36
+}`
+
+const testPodMeta = "pod=default/podName"
+
+func TestRun_FlagValidation(t *testing.T) {
+	cases := []struct {
+		flags  []string
+		expErr string
+	}{
+		{
+			flags:  []string{"-method", testAuthMethod},
+			expErr: "-meta must be set",
+		},
+		{
+			flags:  []string{"-meta", "pod=abcdefg"},
+			expErr: "-method must be set",
+		},
+	}
```

Comment:
I have them taking in defaults right now in the flag parsing, wasn't sure how to handle it, what do you think?

Reply:
I'd say follow the other commands we have and how they deal with flag defaults. If there is no prior art then I'd say if a flag has a default then we don't need to check if it's `nil` and then we also wouldn't write a test for it.

Reply:
Yeah that is what I ended up doing here. but let me look at some other tests and update the PR accordingly.

Reply:
Oh interesting. I would have said that if:
```go
	if c.flagBearerTokenFile == "" {
		c.UI.Error("-bearer-token-file must be set")
		return 1
	}
	if c.flagTokenSinkFile == "" {
		c.UI.Error("-token-sink-file must be set")
		return 1
	}
```

Will never run because those have defaults then we can just delete those lines and then no need to test.